config/initializers/highlight_override.rb

```diff
@@ -59,3 +58,26 @@ def create_model!(user_uuid:)
     Highlight.create!(to_hash.merge(user_uuid: user_uuid))
   end
 end
+
+Api::V0::Bindings::Highlights.class_exec do
+  def self.create_from_model(highlights)
+    new(highlights: highlights.map(&:attributes))
+  end
+end
+
+Api::V0::Bindings::GetHighlights.class_exec do
+  def query(user_uuid:)
+    highlights = ::Highlight.by_user(user_uuid)
+
+    # The submitted GetHighlight properties create automatic chaining via
+    # the by_X scopes on the Highlight model.
+    pagination = to_hash.dup
+    filter_by = pagination.slice!(:page, :per_page)
+    filter_by.each do |key, value|
+      highlights = highlights.public_send("by_#{key}", value) if value.present?
+    end
+
+    highlights.paginate(pagination)
```

Comment:
This blows up if `:page` is not provided.  They are not required, so need to be clear about defaults or not paginate if pagination parameters not provided.  Also I think if pagination is specified we need to also have a default order?  I'm not a fan of default scopes for order defaults fwiw -- always seems to be problematic at some point -- would be happier with a default here if pagination used.

Reply:
i added in defaults for pagination.